frontend/src/views/ScheduleView.vue

```diff
@@ -107,6 +106,8 @@ const onDateChange = (dateObj) => {
   const start = dj(dateObj.start);
   const end = dj(dateObj.end);
 
+  activeDate.value = start.add(end.diff(start, 'minutes')/2, 'minutes');
```

Comment:
We're now updating the selected date on navigation too, to keep the event generation in sync with the view